simexpal/base.py

```diff
@@ -503,10 +503,15 @@ def get_queue_job_status(self, jobid):
 		return Status(self.queue_queried_jobs.get(jobid, Status.FAILED))
 
 	def writeback_status_cache(self):
-		fd, path = tempfile.mkstemp(dir=self.basedir)
-		with os.fdopen(fd, 'w') as tmp:
-			json.dump(self.status_cache_dict, tmp)
-		os.rename(path, self.status_cache_path)
+		#fd, path = tempfile.mkstemp(dir=self.basedir)
+		#with os.fdopen(fd, 'w') as tmp:
+		#	json.dump(self.status_cache_dict, tmp)
+		#os.rename(path, self.status_cache_path)
+		with open(self.basedir + '/simex.cache', 'r') as myfile:
+			cache = json.load(myfile)
+			cache["status"] = self.status_cache_dict
+		with open(self.basedir + '/simex.cache', 'w') as myfile:
```

Comment:
Why do we need to open the file once with read and then with write access? Why not open it with read write access, load the cache from file, write the cache from memory to the read object `cache` object from file and then write it back to file? Or am I not following what happens here?